src/components/mermaid-diagram.tsx

```
@@ -33,12 +37,63 @@ const MermaidChart = ({ chart }: MermaidChartProps) => {
+    // Store ref value for cleanup
+    const currentRef = containerRef.current;
+
+    return () => {
+      const svgElement = currentRef?.querySelector("svg");
+      if (svgElement) {
+        try {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+          // svgPanZoom(svgElement).destroy();
+        } catch (error) {
+          console.error("Failed to destroy pan-zoom instance:", error);
+        }
```

Biome (1.9.4)

[error] This code is unreachable



(lint/correctness/noUnreachable)